src/client/app/app.component.ts

```diff
@@ -1,13 +1,14 @@
 import { Component } from '@angular/core';
 import { ROUTER_DIRECTIVES, Routes } from '@angular/router';
+import { HTTP_PROVIDERS} from '@angular/http';
 
 import { AboutComponent } from './+about/index';
 import { HomeComponent } from './+home/index';
 import { NameListService, NavbarComponent, ToolbarComponent } from './shared/index';
 
 @Component({
   selector: 'sd-app',
-  viewProviders: [NameListService],
+  viewProviders: [NameListService, HTTP_PROVIDERS],
```

Comment:
Just for the sake of consistency (and probably best practice), I'd place `HTTP_PROVIDERS` in `main.ts` together with the router ones.